File: lambda/api/app/schemas/ocr.py
from pydantic import BaseModel
from typing import Optional, List, Dict


class OcrWord(BaseModel):
    """OCR認識された単語の情報"""
    id: int
    content: str
    rec_score: Optional[float] = None
    points: Optional[List[List[float]]] = None
    page: Optional[int] = None
    direction: Optional[str] = None


class OcrResult(BaseModel):
    """OCR処理結果"""
    words: List[OcrWord]
    text: Optional[str] = None
    word_count: Optional[int] = None
    total_pages: Optional[int] = None
    pages: Optional[List[Dict]] = None
    error: Optional[str] = None


class OcrResultResponse(BaseModel):
    """OCR結果取得APIのレスポンス"""
    filename: Optional[str]
    s3_key: Optional[str]
    uploadTime: Optional[str]
    status: Optional[str]
    ocrResult: OcrResult
    imageUrl: Optional[str]


class OcrStartRequest(BaseModel):
    """OCR処理開始リクエスト"""
    app_name: Optional[str] = None

File: web/src/types/ocr.ts
export interface ImageFile {
  id: string;
  name: string;
  s3_key: string;
  uploadTime: string;
  status: 'uploading' | 'pending' | 'processing' | 'completed' | 'failed';
  jobId?: string;
  appName?: string;
  pageProcessingMode?: 'combined' | 'individual';
  pageNumber?: number;
  totalPages?: number;
  parentDocumentId?: string;
  verificationCompleted?: boolean;
}

export interface OcrWord {
  content: string;
  rec_score?: number;
  det_score?: number;
  points?: number[][];
  id?: number;
  page?: number;  // ページ番号を追加（マルチページ対応）
}

export interface OcrResultData extends OcrWord {
  // 拡張プロパティがあれば追加
}

export interface OcrBoundingBox {
  id: number;
  top: number;
  left: number;
  width: number;
  height: number;
  text: string;
}

export interface OcrResponse {
  filename: string;
  s3_key: string;
  uploadTime: string;
  status: string;
  ocrResult: {
    words: OcrWord[];
  };
  imageUrl: string;
  app_name?: string;
}

export interface OcrStartResponse {
  jobId: string;
}

export interface PresignedUrlResponse {
  presigned_url: string;
  s3_key: string;
  image_id: string;
}

export interface UploadCompleteResponse {
  status: string;
  message: string;
  image_id: string;
  is_converting?: boolean;
  page_processing_mode?: 'combined' | 'individual';
}
export interface OcrStatusResponse {
  status: string;
  images: {
    id: string;
    filename: string;
    status: string;
  }[];
}

export interface PresignedDownloadUrlResponse {
  presigned_url: string;
  presigned_urls: Array<{
    page: number;
    presigned_url: string;
    s3_key: string;
  }>;
  total_pages: number;
  is_multipage: boolean;
  content_type: string;
  filename: string;
  is_converted: boolean;
}
